Module-3/README.md

```diff
@@ -0,0 +1,144 @@
+> # This README contains the answers to Module 3's homework questions.
+
+## Part 1 - Short-Answer Questions
+
+
+#### _a) 0...5 and 0..<5 are two types of ?????. How are they different?_
+
+`0...5` is a `ClosedRange<Int>`. It contains values from the lower bound (0) up to and including the upper bound (5). Here, we could iterate over the sequence of contained values: `0, 1, 2, 3, 4, 5`.
+
+`0..<5` is a `Range<Int>` (half-open range). It includes values from the lower bound up to, but not including, the upper bound. Here, we could iterate over the sequence of contained values: `0, 1, 2, 3, 4`.
+
+#### _b) Describe type inference in Swift and give an example._
+
+Type inference in Swift is the compiler's ability to deduce the type of a variable or constant without an explicit type annotation.
+
+*Explicit Type Annotation*:
+```swift
+let name: String = "Mark"
+```
+*Type Inference* (here, Swift infers the type of name to be `String` based on the assigned value):
+```swift
+let name = "Mark"
+```
+Explicit type annotation may be required when Swift cannot unambiguously infer the type, or when you want to specify a type that is different from the one that would be inferred.
+
+#### _c) List three advantages of Playgrounds._
+
+1. Playgrounds can automatically execute code, with the results quickly available for review.
+2. The results of each line of code can be visualized in the sidebar, making it easier to understand the effects of code changes in real-time.
+3. Playgrounds provide a lightweight and convenient environment for experimentation and testing outside of a full project context.
+
+#### _d) When does the execution of a while loop end?_
+
+A while loop ends immediately at the point when the comparison statement evaluates to `false`.
+
+Consider the following example:
+```swift
+var t = 0
+
+while t < 10 {
+    print(t)
+    t += 1
+}
+```
+
+Here, when `t` is incremented to `10`, and the loop comparison is next evaluated, the execution of the while loop will end. This means that while `10` will never be printed to the console (as the loop exits before printing `t` inside of the loop), the value of `t` at which the loop exits will be `10`.
+
+#### _e) True or False: Tuples in Swift can contain values of different data types._
+
+True - tuples can contain values of different data types.
+
+However, once a specific tuple is defined, the types of data it contains remain fixed (immutabe) throughout its lifetime.
+
+```swift
+var locationTuple = (houseNumber: 123, street: "Sesame")
+
+locationTuple.houseNumber = 345     // Works because it's the same type `Int`
+locationTuple.houseNumber = "12/3A" // Would not work, as it's trying to assign a `String` to `houseNumber`, which is of type `Int`
+```
+
+#### _f) List three data types you have used in Swift._
+
+- `String`
+- `Bool`
+- `Int`
+
+#### _g) To execute alternative code when the condition of an if statement is not met, you can use what clause?_
+
+`else` clause
```

Comment:
Let's also not forget about the `else if` statement. ;)

Reply:
Noted.  I would consider an `else if` to still be an `if` statement.  My logic thus read the question as `any if` (`if` or `else if`) ...